scripts/mkcloud

```diff
@@ -774,9 +776,57 @@ function testsetup()
     onadmin testsetup
     local ret=$?
     $scp root@$adminip:tempest.log "$artifacts_dir/tempest.log"
+
+    # Register the cloud in Rally and import the results
+    if [ -n "$rally_server" ]; then
+        local title=$(testname)
+        local type=$(testtype)
+
+        $scp root@$adminip:.openrc "$artifacts_dir/.openrc"
+        local tmpdir=`ssh $sshopts rally@$rally_server "mktemp -d rally-XXXXXX"`
+        $scp "$artifacts_dir/tempest.log" "$artifacts_dir/.openrc" rally@$rally_server:$tmpdir
+        ssh $sshopts rally@$rally_server "
+            source rally/bin/activate
+            source $tmpdir/.openrc
+            deployment=\$(rally deployment create --fromenv --name=\"$title\" | awk '/Using deployment:/{print \$3}')
+            rally verify start \"$type\" --tempest-log=$tmpdir/tempest.log --deployment=\$deployment
+            rm -fr $tmpdir
+        "
+    fi
+
     return $ret
 }
 
+function testname()
+{
+    local name="$cloudsource/ha:$hacloud/ceph:$cephvolumenumber/cinder:$cinder_backend"
+
+    # Check if this instance of mkcloud is running inside Jenkins
+    if [ -n "$JENKINS_URL" ] ; then
+        name="$BUILD_NUMBER - $JOB_NAME - $BUILD_URL"
+    fi
+
+    echo $name
+}
+
+function testtype()
+{
+    # If `tempestopstions` is not set, or contains `smoke`, is a smoke
+    # test, in other case is a full or ad-hoc tesmpest test
+    local type="smoke"
+    if [ -n "$tempestoptions" ]; then
+        if [[ $tempestoptions =~ .*smoke.* ]]; then
+            type="smoke"
+        elif [ "$tempestoptions" = "--subunit" ]; then
```

Comment:
I am not convinced of it.  Like 90% of the code is using `if [ -n "$VAR"] ;`.  But I will change it.